pkg/container/ring_buffer_test.go

```diff
@@ -0,0 +1,206 @@
+package container
```

Comment:
I'm idly wondering if the new fuzz testing framework would be useful here - could we fuzz buffer operations?

Reply:
Yeah I think so, I am a bit concerned about the complexity of this buffer implementation being correct - hence all the tests.